release/pkg/manager/calico/manager.go

```diff
@@ -39,9 +39,10 @@ import (
 // Global configuration for releases.
 var (
 	// Default defaultRegistries to which all release images are pushed.
+	// The first registry is what the manifests use. Currently it is quay.io/calico.
```

Comment:
This isn't what I expected - I'd have expected that `gen-manifests` should use the values specified in values.yaml unless the user has provided an override. 

I think we should keep that behavior - otherwise, the CI checks that verify manifests are up-to-date are testing the wrong thing.

Reply:
this does not negate the CI checks that verify manifests.

Below, the first registry in image registries is used to generate the manifests. If a user does not override the default, this is stated to keep it aligned with what happens when `gen-manifests` is run separately.

Reply:
> keep it aligned with what happens when gen-manifests is run separately

Keeping something aligned / in-sync is not nearly as good as running it through the same code path.

If `values.yaml` changes, this code will need to be updated as well to match it - and if for whatever reason we forget to do that, it will lead to problems. 

I think the desired behavior is:
- Use the value from values.yaml by default
- If a dev has specified a custom registry, use that instead.